source/h1_connection.c

```diff
@@ -985,7 +992,16 @@ static int s_decoder_on_header(const struct aws_h1_decoded_header *header, void
     /* RFC-7230 section 6.1.
      * "Connection: close" header signals that a connection will not persist after the current request/response */
     if (header->name == AWS_HTTP_HEADER_CONNECTION) {
-        if (aws_byte_cursor_eq_c_str(&header->value_data, "close")) {
+        bool ignore_connection_close = false;
```

Comment:
You can do all the voodoo right here without any extra variables or locking:

```
/* for bug for bug compatibility with L7 proxies. They can send connection:close on successful connect
 * responses. This should be ignored. */
bool ignore_connection_close =
        incoming_stream->base.request_method == AWS_HTTP_METHOD_CONNECT
        && incoming_stream->client_data
        && incoming_stream->client_data->response_status == AWS_HTTP_STATUS_200_OK;
```
Feel free to rewrite that however looks good in your eyes (and in the eyes of clang-format)

You should define AWS_HTTP_METHOD_CONNECT and AWS_HTTP_STATUS_200_OK. I'd been trying to only define enums for things we're actually acting on. There are a few steps to hooking up the method enum, just do all the same steps as the others.